templates/default/components/Dropdown.js

```diff
@@ -0,0 +1,90 @@
+import { useState, Fragment, useRef } from 'react';
+import { Menu, Transition } from '@headlessui/react';
+import Link from 'next/link';
+import { MyLink } from './MyLink';
+
+function Dropdown({ item, router }) {
```

Comment:
Just by looking at this function signature, I'm thinking:
1) Why would a dropdown element need to be passed a router? It shouldn't really have anything to do with it. If you think about it its job is to simply take an item with its subitems and display them. That's it 😉. If you do need to pass a current location to it (which doesn't really make sense if it's called `Dropdown`, cuz again, why would a dropdown need a current location?), you should get this value in the parent component and pass only this information (location) to this component, as this is the only part relevant to it.
2) Ok, so it's a dropdown. I imagine the `item` prop holds an item with some child items to display in the dropdown list. And it sounds to me like I could use it to list almost anything. But actually, it's a very specific kind of dropdown. It wraps all the subitems in anchor tags. So it's rather a `MenuItem` or `NavItem` than just a `Dropdown`. Of course, in the future, if we needed a simple dropdown component as well we would extract the dropdown functionality to some `Dropdown` component, and your component (say `NavItem`) would use it internally. But for now, let's just rename it 😉